src/dhtproto/client/request/Update.d

```diff
@@ -0,0 +1,129 @@
+/*******************************************************************************
+
+    Client DHT Update request definitions / handler.
+
+    The Update request attempts to read one record from the specified DHT
+    channel and replace its value in the DHT with a modified version, specified
+    by the user. This works as follows:
+        1. The client selects a connected DHT node from its registry, based on
+           the key of the record to be updated.
+        2. A request is sent to the selected node, asking for the specified
+           record in the specified channel to be returned.
+        3. If the record exists, its value is passed to the user, who may
+           provide a modified version of the value or decide to leave the value
+           as it is.
+        4. The modified version of the value is then sent back to the DHT. One
+           of the following then happens:
+            4a. The record has been modified by another request, in the
+                meantime. The Update request is rejected and has no effect. (The
+                user may retry the request.)
+            4b. The record has not been modified, in the meantime, and the new
+                value is written to the DHT.
```

Comment:
`The record was deleted or has not been modified`, to be pedant.

Reply:
Mm, which sentence is this an addition to?

Reply:
I forgot how the code looked like (I need to check), but AFAIR new value gets written in two cases:

1. the old record value is still valid
2. record ceased to exist in the meantime